### ISSUE TITLE
[BUG] UnsatisfiedLinkError in librocksdbjni-win64.dll from rocksdbjni-6.6.4.jar

### ISSUE THREAD

**Issue by stefan-zobel** [facebook/rocksdb#6480]
> Note: Please use Issues only for bug reports. For questions, discussions, feature requests, etc. post to dev group: https://groups.google.com/forum/#!forum/rocksdb or https://www.facebook.com/groups/rocksdb.dev

### Expected behavior
The basic example from https://github.com/facebook/rocksdb/wiki/RocksJava-Basics#opening-a-database should work without errors.

### Actual behavior
```
Exception in thread "main" java.lang.UnsatisfiedLinkError: org.rocksdb.RocksDB.closeDatabase(J)V
	at org.rocksdb.RocksDB.closeDatabase(Native Method)
	at org.rocksdb.RocksDB.close(RocksDB.java:464)
```

### Steps to reproduce the behavior
Try to run https://github.com/facebook/rocksdb/wiki/RocksJava-Basics#opening-a-database on Windows 10 using **_rocksdbjni-6.6.4.jar_** from Maven Central

### Details:

When using rocksdbjni-**_6.6.4_**.jar from Maven Central to run the basic example from https://github.com/facebook/rocksdb/wiki/RocksJava-Basics#opening-a-database I get 

```
Exception in thread "main" java.lang.UnsatisfiedLinkError: org.rocksdb.RocksDB.closeDatabase(J)V
	at org.rocksdb.RocksDB.closeDatabase(Native Method)
	at org.rocksdb.RocksDB.close(RocksDB.java:464)
```

This is on Windows 10. When I use the earlier rocksdbjni-**_6.5.3_**.jar from Maven Central everything works fine.

I've looked at the exported symbols in **_librocksdbjni-win64.dll_** for both versions and observed that the **_6.6.4_** version indeed misses symbols.

For example, closeDatabase(J)V from org.rocksdb.**TransactionDB** is also missing:

```
Exception in thread "main" java.lang.UnsatisfiedLinkError: org.rocksdb.TransactionDB.closeDatabase(J)V
	at org.rocksdb.TransactionDB.closeDatabase(Native Method)
	at org.rocksdb.TransactionDB.close(TransactionDB.java:145)
```

### Number of exported sysmbols in 6.6.4 versus 6.5.3

- rocksdbjni-**6.6.4**.jar: librocksdbjni-win64.dll has **1260** exported symbols overall (Java_org_rocksdb_RocksDB has **76** exported symbols, ordinal 704 to 779)

- rocksdbjni-**6.5.3**.jar: librocksdbjni-win64.dll has **1450** exported symbols overall (Java_org_rocksdb_RocksDB has **99** exported symbols, ordinal 815 to 913)

I suspect something did go wrong during the generation of the header files when the 6.6.4 release was built.

**Comment by stefan-zobel** [comment]
Not a single comment within 8 days. That's telling.

**Comment by adamretter** [comment]
@stefan-zobel sorry for the slow response. I did not see any initial notification. This is the first I have seen about this.

I would have built the RocksJava 6.6.4 release here. I will look into this later today...

Can you tell me, is there a simple command you ran to count the number of exported symbols?

**Comment by stefan-zobel** [comment]
Hi Adam, thanks for looking into this.
I counted the exported symols simply by using a 2006 version of Dependency Walker.

**Comment by kosiakk** [comment]
My Windows 7 machine has nearly every MS Visual C++ runtime installed, but it fails to start:
```
Exception in thread "main" java.lang.UnsatisfiedLinkError: C:\Users\tk8k1\AppData\Local\Temp\librocksdbjni723692463416895557.dll: The operating system cannot run %1
	at java.base/java.lang.ClassLoader$NativeLibrary.load0(Native Method)
	at java.base/java.lang.ClassLoader$NativeLibrary.load(ClassLoader.java:2440)
	at java.base/java.lang.ClassLoader$NativeLibrary.loadLibrary(ClassLoader.java:2497)
	at java.base/java.lang.ClassLoader.loadLibrary0(ClassLoader.java:2694)
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2627)
	at java.base/java.lang.Runtime.load0(Runtime.java:744)
	at java.base/java.lang.System.load(System.java:1873)
	at org.rocksdb.NativeLibraryLoader.loadLibraryFromJar(NativeLibraryLoader.java:79)
	at org.rocksdb.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:57)
	at org.rocksdb.RocksDB.loadLibrary(RocksDB.java:63)
	at org.rocksdb.RocksDB.<clinit>(RocksDB.java:34)
	at org.rocksdb.Options.<clinit>(Options.java:27)
```

I'm not even sure how to debug this problem

**Comment by adamretter** [comment]
@stefan-zobel I can reproduce your issue, but I can't really say what happened with the Windows library for RocksJava 6.4.6. As artifacts published to Maven Central are immutable, I also can't fix it directly.

I have just released version 6.7.3 to Maven Central (after checking that it doesn't have the same problem). I would suggest your best bet is to upgrade to that version.
